diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt

```diff
@@ -0,0 +1,242 @@
+import org.cqfn.diktat.common.config.rules.DIKTAT_ANALYSIS_CONF
+import org.cqfn.diktat.util.isKotlinCodeOrScript
+import org.cqfn.diktat.util.tryToPathIfExists
+import org.cqfn.diktat.util.walkByGlob
+import generated.DIKTAT_VERSION
+import generated.KTLINT_VERSION
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.LoggerContext
+import org.slf4j.event.Level
+import java.io.OutputStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.createDirectories
+import kotlin.io.path.outputStream
+import kotlin.system.exitProcess
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import kotlinx.cli.vararg
+
+/**
+ * @property config path to `diktat-analysis.yml`
+ * @property mode mode of `diktat`
+ * @property reporterProviderId
+ * @property output
+ * @property patterns
+ */
+data class DiktatProperties(
+    val config: String,
+    val mode: DiktatMode,
+    val reporterProviderId: String,
+    val output: String?,
+    private val groupByFileInPlain: Boolean,
+    private val colorNameInPlain: String?,
+    private val logLevel: Level,
+    val patterns: List<String>,
+) {
+    /**
+     * Configure logger level using [logLevel]
+     */
+    fun configureLogger() {
+        // set log level
+        LogManager.getContext(false)
+            .let { it as LoggerContext }
+            .also { ctx ->
+                ctx.configuration.rootLogger.level = when (logLevel) {
+                    Level.ERROR -> org.apache.logging.log4j.Level.ERROR
+                    Level.WARN -> org.apache.logging.log4j.Level.WARN
+                    Level.INFO -> org.apache.logging.log4j.Level.INFO
+                    Level.DEBUG -> org.apache.logging.log4j.Level.DEBUG
+                    Level.TRACE -> org.apache.logging.log4j.Level.TRACE
+                }
+            }
+            .updateLoggers()
+    }
+
+    /**
+     * @param sourceRootDir
+     * @param loggingListener
+     * @return [DiktatRunnerArguments] created from [DiktatProperties]
+     */
+    fun toRunnerArguments(
+        sourceRootDir: Path,
+        loggingListener: DiktatProcessorListener,
+    ): DiktatRunnerArguments = DiktatRunnerArguments(
+        configFileName = config,
+        sourceRootDir = sourceRootDir,
+        files = getFiles(sourceRootDir),
+        baselineFile = null,
+        reporterType = reporterProviderId,
+        reporterOutput = getReporterOutput(),
+        groupByFileInPlain = groupByFileInPlain,
+        colorNameInPlain = colorNameInPlain,
+        loggingListener = loggingListener,
+    )
+
+    private fun getFiles(sourceRootDir: Path): Collection<Path> = patterns
+        .asSequence()
+        .flatMap { pattern ->
+            pattern.tryToPathIfExists()?.let { sequenceOf(it) }
+                ?: sourceRootDir.walkByGlob(pattern)
+        }
+        .filter { file -> file.isKotlinCodeOrScript() }
+        .distinct()
```

Comment:
I'd put `distinct()` _after_ `toAbsolutePath()` and `normalize()` are called.

Reply:
done